autoload/airline/extensions/wordcount.vim

```diff
@@ -4,7 +4,13 @@
 scriptencoding utf-8
 
 " get wordcount {{{1
-if exists('*wordcount')
+if exists('*vimtex#misc#wordcount')
+" We're in a TeX file and vimtex is a plugin, so use it's wordcount...
```

Comment:
No, that's not true. The function will exist even if you are not in a TeX file, as long as you have VimTeX properly installed. So, it would probably be better to use `if exists('b:vimtex')` or `if &filetype ==# 'tex' && exists('*vimtex#misc#wordcount')`. I believe the former should suffice and is more simple.

Reply:
Or, perhaps `if &filetype ==# 'tex' && exists('b:vimtex')` (because in some edge cases, the `b:vimtex` will exist even for non LaTeX files).

Reply:
Okay, so now I'm confused.  First, thanks for checking in on this.  

Actually my original coding (for whatever reason) seems to work just fine if you have only one type of file in the editor, (like latex, or markdown), and it switches between the alternate wordcounts, as far as I can tell, seamlessly.  

The problem arises when you say open two files with alternate wordcount versions, and then there is much consternation in vim.  Right now I just pushed a version (to the tunaboy/vim-airline repo) with the 1st version  and iv I open up just a tex file it works.  And a markdown file it works (with the markdown wordcount), but open both at once and then it goes nuts.

This may be an internal to airline, as it assumes there is only one function defined?  Yes it is.
With the tunaboy/vim-airline (my suggestion, with your correct coding, 1st version of your setup):

    vim -O file.tex file.md

Causes the vim complaints. But if I do:

   vim -O file.md file.tex

Then vim does not complain, but the wordcount for the TeX buffer is 0 (and the markdown buffer wordcount looks right)

Reply:
If the airline folks can chime in, and maybe my simplistic hack is not suitable, but there's another way to do this...

Reply:
I really like having the correct wordcount for latex files in my status line.

Reply:
I have to leave this for a bit, but your coding looks perfect.  But I'm obviously not understanding the structure of vim-airline, which is possibly one of the bigger understatements of the year.  

There are issues with b:vimtex not being defined, but as far as I can see, it only gets used in the exists('b:vimtex') bit.

I have to leave this for a bit, but I appreciate your suggestion. It seems the bulletproof way to do this.

Reply:
And now I've pushed a version that works with single tex files

    vim file.tex

with mixed files that have different wordcount functions

     vim file.tex file.md

and in a single window with the two filetypes mixed together

     vim -O file.tex file.md

But if the TeX file is the second in this list, the buffer needs to be reloatded (switch to that buffer and :e) and then the wordcount appears.  

Have to again leave this for a bit, (to write the abstract that caused me to want this accurate wordcount.)

Happy Halloween folks!   (though you may be at the end of that day?)  

Cheers, all!

Reply:
You are now responding to my comment, which makes it much harder to follow this thread. I would propose that you mark this thread resolved, then give a brief update (like the one here) in the main thread.

I don't use airline, so I can't help out with that part.